tests/testsuite/rustflags.rs

```diff
@@ -272,6 +272,47 @@ fn env_rustflags_build_script_with_target() {
         .run();
 }
 
+#[cargo_test]
+fn env_rustflags_build_script_with_target_and_applies_to_host_kind() {
+    // RUSTFLAGS *should* be passed to rustc for build scripts when --target is specified as the
+    // host triple and target-applies-to-host-kind is enabled.
```

Comment:
This also looks wrong, `target-applies-to-host` was supposed to be for controlling target table config applicability to build scripts and such, not anything else.

Reply:
The same argument applies here — the stated intention for `RUSTFLAGS` is that it applies "to the current target", and if `--target <host>` is passed and `target-applies-to-host = true`, then that implies that `RUSTFLAGS` should also apply to the host.

Reply:
The issue is retaining bug for bug compatibility when `target-applies-to-host = true` is set. It wasn't intended that `target-applies-to-host = true` do anything until after the default change.